sgx-types/src/quote.rs

```diff
@@ -0,0 +1,250 @@
+    /// PCK Leaf Cert||Intermediate CA Cert||Root CA Cert
+    PCKCertChain = 5,
+
+    /// Intel SGX Quote (not supported).
+    Quote = 6,
+
+    /// Platform Manifest (not supported).
+    Manifest = 7,
+}
+
+impl Default for CertDataType {
+    fn default() -> Self {
+        Self::PCKCertChain
+    }
+}
+
+/// Data required to verify the QE Report Signature.
+/// Section A.4, Table 9
+#[repr(C)]
+pub struct QECert {
+    certdatatype: u16,
+    certdata: [u8; 4096],
+}
+
+impl Default for QECert {
+    fn default() -> Self {
+        Self {
+            certdatatype: Default::default(),
+            certdata: [0u8; 4096],
+        }
+    }
+}
+
+/// ECDSA  signature, the r component followed by the
+/// s component, 2 x 32 bytes.
+/// A.4, Table 6
+#[derive(Default)]
+#[repr(C)]
+pub struct ECDSAP256Sig {
+    /// r component
+    pub r: [u8; 32],
+
+    /// s component
+    pub s: [u8; 32],
+}
+
+/// EC KT-I Public Key, the x-coordinate followed by
+/// the y-coordinate (on the RFC 6090P-256 curve),
+/// 2 x 32 bytes.
+/// A.4, Table 7
+#[derive(Default)]
+#[repr(C)]
+pub struct ECDSAPubKey {
+    /// x coordinate
+    pub x: [u8; 32],
+
+    /// y coordinate
+    pub y: [u8; 32],
+}
+
+/// A.4, Table 4
+#[derive(Default)]
+#[repr(C)]
+pub struct SigData {
+    isv_enclave_report_sig: ECDSAP256Sig,
+    ecdsa_attestation_key: ECDSAPubKey,
+    qe_report: Body,
+    qe_report_sig: ECDSAP256Sig,
+    qe_auth: QEAuth,
+    qe_cert: QECert,
+}
+
+/// The type of Attestation Key used to sign the Report.
+#[repr(u16)]
+pub enum AttestationKeyType {
+    /// ECDSA-256-with-P-256 curve
+    ECDSA256P256 = 2,
+
+    /// ECDSA-384-with-P-384 curve; not supported
+    ECDSA384P384 = 3,
+}
+
+impl Default for AttestationKeyType {
+    fn default() -> Self {
+        AttestationKeyType::ECDSA256P256
+    }
+}
+
+/// Unlike the other parts of the Quote, this structure
+/// is transparent to the user.
+/// Section A.4, Table 3
+#[repr(C)]
+pub struct QuoteHeader {
+    /// Version of Quote structure, 3 in the ECDSA case.
+    pub version: u16,
+
+    /// Type of attestation key used. Only one type is currently supported:
+    /// 2 (ECDSA-256-with-P-256-curve).
+    pub att_key_type: AttestationKeyType,
+
+    /// Reserved.
+    reserved: u32,
+
+    /// Security version of the QE.
+    pub qe_svn: u16,
+
+    /// Security version of the Provisioning Cerfitication Enclave.
+    pub pce_svn: u16,
+
+    /// ID of the QE vendor.
+    pub qe_vendor_id: [u8; 16],
+
+    /// Custom user-defined data. For the Intel DCAP library, the first 16 bytes
+    /// contain a QE identifier used to link a PCK Cert to an Enc(PPID). This
+    /// identifier is consistent for every quote generated with this QE on this
+    /// platform.
+    pub user_data: [u8; 20],
+}
+
+impl Default for QuoteHeader {
+    fn default() -> Self {
+        let mut user_data = [0u8; 20];
+        user_data.copy_from_slice(&INTELVID); // TODO: Endianness of byte arrays?
```

Comment:
I don't think we need to worry about endianness here. Individual bytes (and I think arrays of) are endian-independent. I also think that all of Intel's chips are little endian.

Reply:
The thing I'm concerned about is whether the bytes themselves need to be ordered as BE or LE, because I didn't see that documented anywhere. I just wouldn't want to record the number backwards. Then again, I think `user_data` is the most flexible field here, so maybe it is fine. It depends on whether this needs to match anything that we don't control, and I can't remember right now whether that's the case.

Reply:
@lkatalin I'm with @connorkuehl here. I don't think we need to worry about it. If we find something wrong, we'll discover it later.

Reply:
@connorkuehl @npmccallum I was thinking of something like this: https://doc.rust-lang.org/std/primitive.u32.html#method.to_le_bytes But I am more than happy to drop it and not worry about it.